lib/User_Proxy.php

```diff
@@ -27,13 +27,18 @@
 
 namespace OCA\User_LDAP;
 
-use OCA\User_LDAP\User\User;
 use OCP\IConfig;
 use OCP\IUserBackend;
+use OCP\User\IProvidesEMailBackend;
 use OCP\User\IProvidesExtendedSearchBackend;
+use OCP\User\IProvidesQuotaBackend;
 use OCP\UserInterface;
 
-class User_Proxy extends Proxy implements IUserBackend, UserInterface, IProvidesExtendedSearchBackend {
+class User_Proxy extends Proxy implements IUserBackend, UserInterface, IProvidesQuotaBackend, IProvidesExtendedSearchBackend, IProvidesEMailBackend {
```

Comment:
I'm forced to complain about the massive interface implementation here, but no change needed for now because it will require quite huge changes, and not just here.

Reply:
The app used to update core itself, which is not the way it shoud be. This PR just corrects that by removing the related code and implementing the interfaces to get the correct behavior.